File: CS/LongRunningOperations.Module/BusinessObjects/Object1.cs
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.ConditionalAppearance;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace LongRunningOperations.Module.BusinessObjects {
    [DefaultClassOptions]
    [Appearance("IsInProcessing", "[IsInProcessing]='True'", Enabled = false, TargetItems="*")]
    public class Object1 : BaseObject {
        internal bool? isInProcessing; 
        private string name;
        private string description;
        public string Name {
            get { return name; }
            set {
                SetPropertyValue<string>("Name", ref name, value);
            }
        }
        [Size(SizeAttribute.Unlimited)]
        public string Description {
            get { return description; }
            set {
                SetPropertyValue<string>("Description", ref description, value);
            }
        }
        public bool IsInProcessing {
            get {
                if(!isInProcessing.HasValue) {
                    isInProcessing = (Session.FindObject<ObjectForLongRunningOperations>(new BinaryOperator("ObjectToProcess", this)) != null);
                }
                return isInProcessing.Value; 
            }
        }
        public Object1(Session session)
            : base(session) {

        }
        public void SetIsInProcessing(bool val) {
            isInProcessing = val;
            RaisePropertyChangedEvent("IsInProcessing");
        }
    }
}

File: VB/LongRunningOperations.Module/BusinessObjects/Object1.vb
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.ConditionalAppearance
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

Namespace LongRunningOperations.Module.BusinessObjects
	<DefaultClassOptions>
	<Appearance("IsInProcessing", "[IsInProcessing]='True'", Enabled := False, TargetItems:="*")>
	Public Class Object1
		Inherits BaseObject

'INSTANT VB NOTE: The field isInProcessing was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Friend isInProcessing_Renamed? As Boolean
'INSTANT VB NOTE: The field name was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Private name_Renamed As String
'INSTANT VB NOTE: The field description was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Private description_Renamed As String
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Name", name_Renamed, value)
			End Set
		End Property
		<Size(SizeAttribute.Unlimited)>
		Public Property Description() As String
			Get
				Return description_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Description", description_Renamed, value)
			End Set
		End Property
		Public ReadOnly Property IsInProcessing() As Boolean
			Get
				If Not isInProcessing_Renamed.HasValue Then
					isInProcessing_Renamed = (Session.FindObject(Of ObjectForLongRunningOperations)(New BinaryOperator("ObjectToProcess", Me)) IsNot Nothing)
				End If
				Return isInProcessing_Renamed.Value
			End Get
		End Property
		Public Sub New(ByVal session As Session)
			MyBase.New(session)

		End Sub
		Public Sub SetIsInProcessing(ByVal val As Boolean)
			isInProcessing_Renamed = val
			RaisePropertyChangedEvent("IsInProcessing")
		End Sub
	End Class
End Namespace